core/net/mac/tsch/tsch-security.h

```diff
@@ -39,22 +39,36 @@
 #include "net/mac/tsch/tsch-asn.h"
 #include "net/mac/tsch/tsch-private.h"
 #include "net/mac/frame802154.h"
+#include "net/llsec/llsec802154.h"
 #include "net/mac/frame802154e-ie.h"
 
 /******** Configuration *******/
 
 /* To enable TSCH security:
- * - set LLSEC802154_CONF_SECURITY_LEVEL
+ * - set LLSEC802154_CONF_ENABLED
+ * - set TSCH_SECURITY_CONF_LEVEL
  * - set LLSEC802154_CONF_USES_EXPLICIT_KEYS
  * - unset LLSEC802154_CONF_USES_FRAME_COUNTER
  * */
-#define TSCH_SECURITY_ENABLED (LLSEC802154_CONF_SECURITY_LEVEL != 0)
-#if TSCH_SECURITY_ENABLED && !LLSEC802154_CONF_USES_EXPLICIT_KEYS
-#error TSCH_SECURITY_ENABLED set but LLSEC802154_CONF_USES_EXPLICIT_KEYS unset
-#endif /* TSCH_SECURITY_ENABLED */
-#if TSCH_SECURITY_ENABLED && LLSEC802154_CONF_USES_FRAME_COUNTER
-#error TSCH_SECURITY_ENABLED set but LLSEC802154_CONF_USES_FRAME_COUNTER set
-#endif /* TSCH_SECURITY_ENABLED */
+
+#ifdef TSCH_SECURITY_CONF_LEVEL
+#define TSCH_SECURITY_LEVEL TSCH_SECURITY_CONF_LEVEL
+#else /* TSCH_SECURITY_CONF_LEVEL */
+#define TSCH_SECURITY_LEVEL FRAME802154_SECURITY_LEVEL_NONE
+#endif /* TSCH_SECURITY_CONF_LEVEL */
```

Comment:
I don't think we need such a flag at all: TSCH security levels (and keys and key indexes) are defined in https://github.com/contiki-os/contiki/blob/master/core/net/mac/tsch/tsch-security.h

We used `LLSEC802154_CONF_SECURITY_LEVEL` so far only in a binary way (sec enabled or not), I guess `LLSEC802154_ENABLED` is now perfectly suited for the task.